lms/djangoapps/grades/tasks.py

```diff
@@ -315,14 +316,28 @@ def _update_subsection_grades(
     student = User.objects.get(id=user_id)
     store = modulestore()
     with store.bulk_operations(course_key):
-        course_structure = get_course_blocks(student, store.make_course_usage_key(course_key))
+        course_usage_key = store.make_course_usage_key(course_key)
+        course_structure = get_course_blocks(student, course_usage_key)
         subsections_to_update = course_structure.get_transformer_block_field(
             scored_block_usage_key,
             GradesTransformer,
             'subsections',
             set(),
         )
 
+        # When access is restricted and course blocks are cached without the restricted blocks,
+        # we will update the cache from the store and get the course blocks.
```

Comment:
Maybe a more readable version of this comment could be:

```suggestion
        # Update the cache from the store and get the course blocks if access is restricted and course blocks are 
        # cached without those restricted blocks.
```